smack-experimental/src/main/java/org/jivesoftware/smackx/spoiler/SpoilerManager.java

```diff
@@ -0,0 +1,88 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.spoiler;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.spoiler.element.SpoilerElement;
+
+public final class SpoilerManager extends Manager {
+
+    public static final String NAMESPACE_0 = "urn:xmpp:spoiler:0";
+
+    private static final Map<XMPPConnection, SpoilerManager> INSTANCES = new WeakHashMap<>();
+
+    /**
+     * Create a new SpoilerManager and add Spoiler to disco features.
+     *
+     * @param connection xmpp connection
+     */
+    private SpoilerManager(XMPPConnection connection) {
+        super(connection);
+        ServiceDiscoveryManager.getInstanceFor(connection).addFeature(NAMESPACE_0);
+    }
+
+    /**
+     * Return the connections instance of the SpoilerManager.
+     *
+     * @param connection xmpp connection
+     * @return SpoilerManager
+     */
+    public static SpoilerManager getInstanceFor(XMPPConnection connection) {
+        SpoilerManager manager = INSTANCES.get(connection);
+        if (manager == null) {
+            manager = new SpoilerManager(connection);
+            INSTANCES.put(connection, manager);
+        }
+        return manager;
+    }
+
+    /**
+     * Create an empty SpoilerElement.
+     *
+     * @return empty SpoilerElement
+     */
+    public SpoilerElement createSpoiler() {
```

Comment:
There is no need for those methods to be non-static. It is also common in Smack that those type of methods are part of the SpoilerElement class. I also don't see a reason why the API would need the with-argument variant of the `createSpoiler(…)` method, the user could simply use the constructor. And for the non-argument variant, I would probably go with a static final instance field, .e.g `SpoilerElement.EMPTY_INSTANCE`.